File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/nurikabe/domain/NurikabeObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.nurikabe.domain;

public abstract class NurikabeObject {
    public abstract String objAsString();
    public static NurikabeObject objTypeFromString(String str) {
        switch (str) {
            case "wall":
                return new NurikabeWallObject();
            case "marker":
                return new NurikabeMarkerObject();
            default:
                return new NurikabeEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/nurikabe/NurikabeObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.nurikabe

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class NurikabeObject {
    Empty, Hint, Marker, Wall
}

class NurikabeGameMove(val p: Position, var obj: NurikabeObject = NurikabeObject.Empty)
